xmtp_id/src/associations/verified_signature.rs

```diff
@@ -100,6 +110,59 @@ impl VerifiedSignature {
         ))
     }
 
+    pub fn from_passkey<Text: AsRef<str>>(
+        signature_text: Text,
+        public_key: &[u8],
+        signature: &[u8],
+        authenticator_data: &[u8],
+        client_data_json: &[u8],
+    ) -> Result<Self, SignatureError> {
+        let client_data: ClientDataJson = serde_json::from_slice(client_data_json)
+            .map_err(|_| SignatureError::InvalidClientData)?;
+
+        let signature_text = BASE64_URL_SAFE_NO_PAD.encode(signature_text.as_ref());
+        if signature_text != client_data.challenge {
+            // Challenge needs to match signature text
+            return Err(SignatureError::InvalidClientData);
+        }
+
+        // 1. Parse the public key from raw bytes
+        let verifying_key = match VerifyingKey::from_sec1_bytes(public_key) {
+            Ok(key) => key,
+            Err(_err) => return Err(SignatureError::InvalidPublicKey),
+        };
+
+        // 2. Parse the signature
+        let signature = match Signature::from_der(signature) {
```

Comment:
This would be nicer/more readable as `let signature = Signature::from_der(signature).ok_or(SignatureError::InvalidPublicKey)?`